lib/crypto.js

```diff
@@ -538,8 +538,11 @@ exports.pbkdf2 = function(password,
                           digest,
                           callback) {
   if (typeof digest === 'function') {
-    callback = digest;
-    digest = undefined;
+    internalUtil.deprecate(function() {
+      callback = digest;
+      digest = undefined;
+    }, 'crypto.pbkdf2 without specifying a digest is deprecated. ' +
+       'Please specify a digest')();
```

Comment:
Can you write this as:
```javascript
if (typeof digest === 'function') {
  callback = digest;
  digest = undefined;
  pbkdf2DeprecationWarning();
}
```
Where `pbkdf2DeprecationWarning` is defined outside the function body:
```javascript
const pbkdf2DeprecationWarning =
    internalUtil.deprecate(() => {}, 'crypto.pbkdf2 without...');
```
That avoids a) allocating a closure context for every call and b) printing the deprecation message repeatedly.